pybop/models/empirical/base_ecm.py

```diff
@@ -111,3 +114,86 @@ def _check_params(
             A boolean which signifies whether the parameters are compatible.
         """
         return True
+
+    def get_initial_state(
```

Comment:
This should be updated on PyBaMM first, otherwise we will be out of alignment.

Reply:
Do we use the concatenated `"V"` elsewhere than this method? If not, I'm advocating for the addition of the `"V"` to happen in this method depending on an argument. This would be more robust than inspecting endswith("V") as this breaks with a simple case-change (`"v"`) and contains the string conversion of init_ocv into a single method.

Reply:
The `"V"` is used to flag the type of initial state. I think we should avoid passing the type and value of the initial state separately. We could update the `initial_state` input to a dictionary instead?

Reply:
Yep, I'd be happy with a dictionary into this method with keys to determine type instead.

Reply:
I have refactored the code from `init_soc`/`init_ocv` into `initial_state`. It is somewhat clunkier but should allow for flexibility in the future. I have added warnings to try to guide users away from ill-defined problems that are now explicitly allowed through `initial_state`.